Tests/TestKernel.php

```diff
@@ -0,0 +1,168 @@
+         *
+         * {@inheritdoc}
+         *
+         * @return SymfonyResponse
+         */
+        public function handle(SymfonyRequest $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+        {
+            if (false === $this->booted) {
+                $this->boot();
+            }
+
+            if ($this->getContainer()->has('shop')) {
+                $this->getContainer()->get('shop')->setHost($request->getHost());
+            }
+
+            /** @var $front \Enlight_Controller_Front */
+            $front = $this->container->get('front');
+
+            $front->Response()->setHttpResponseCode(200);
+
+            $request = $this->transformSymfonyRequestToEnlightRequest($request);
+            $front->setRequest($request);
+            $response = $front->dispatch();
+            $response = $this->transformEnlightResponseToSymfonyResponse($response);
+
+            return $response;
+        }
+
+        public function boot($skipDatabase = false)
+        {
+            $result = parent::boot($skipDatabase);
+
+            /** @var $repository \Shopware\Models\Shop\Repository */
+            $repository = $this->getContainer()->get('models')->getRepository('Shopware\Models\Shop\Shop');
+            $shop = $repository->getActiveDefault();
+            @$shop->registerResources();
+
+            return $result;
+        }
+
+        protected function buildContainer()
+        {
+            $containerBuilder = parent::buildContainer();
+
+            $loader = new XmlFileLoader($containerBuilder, new FileLocator(__DIR__));
+            $loader->load('test-services.xml');
+
+            return $containerBuilder;
+        }
+
+        public function free()
+        {
+            if (\Zend_Session::isStarted() && \Zend_Session::isWritable()) {
+                $this->getContainer()->get('session')->unsetAll();
+                \Zend_Session::writeClose();
+
+                unset($_SESSION);
+            }
+
+            $this->getContainer()->get('dbal_connection')->close();
+            $this->getContainer()->reset('dbal_connection');
+
+            \Smarty_Resource::$sources = [];
+            \Smarty_Resource::$compileds = [];
+
+            Shopware(new EmptyShopwareApplication());
+        }
+
+        /**
+         * Terminates a request/response cycle.
+         *
+         * Should be called after sending the response and before shutting down the kernel.
+         *
+         * @param SymfonyRequest $request A Request instance
+         * @param SymfonyResponse $response A Response instance
+         */
+        public function terminate(SymfonyRequest $request, SymfonyResponse $response)
+        {
+            \Smarty_Resource::$sources = [];
+            \Smarty_Resource::$compileds = [];
+
+            $this->container->reset('front');
+            \Enlight_Template_Manager::$_smarty_vars = [];
+        }
+    }
+
+    class EmptyShopwareApplication
```

Comment:
why are here multiple classes in one file?
please split it to different files if it's possible

Reply:
This class is simply a stub to keep the following tests rom using the kernel ater the functional tests ran. It's only used this once and wont be modified in the future, so i think it would just pollute the file system